hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1008 @@
+client_config_sections LOAD_BALANCER("load-balancer", false);
+client_config_sections NEAR_CACHE("near-cache", true);
+client_config_sections QUERY_CACHES("query-caches", false);
+client_config_sections BACKUP_ACK_TO_CLIENT("backup-ack-to-client-enabled", false);
+client_config_sections INSTANCE_NAME("instance-name", false);
+client_config_sections CONNECTION_STRATEGY("connection-strategy", false);
+client_config_sections USER_CODE_DEPLOYMENT("user-code-deployment", false);
+client_config_sections FLAKE_ID_GENERATOR("flake-id-generator", true);
+client_config_sections RELIABLE_TOPIC("reliable-topic", true);
+client_config_sections LABELS("client-labels", false);
+client_config_sections CLUSTER_NAME("cluster-name", false);
+client_config_sections METRICS("metrics", false);
+client_config_sections INSTANCE_TRACKING("instance-tracking", false);
+
+std::vector<client_config_sections> values = { HAZELCAST_CLIENT,
+                                               IMPORT,
+                                               SECURITY,
+                                               PROXY_FACTORIES,
+                                               PROPERTIES,
+                                               SERIALIZATION,
+                                               NATIVE_MEMORY,
+                                               LISTENERS,
+                                               NETWORK,
+                                               LOAD_BALANCER,
+                                               NEAR_CACHE,
+                                               QUERY_CACHES,
+                                               BACKUP_ACK_TO_CLIENT,
+                                               INSTANCE_NAME,
+                                               CONNECTION_STRATEGY,
+                                               USER_CODE_DEPLOYMENT,
+                                               FLAKE_ID_GENERATOR,
+                                               RELIABLE_TOPIC,
+                                               LABELS,
+                                               CLUSTER_NAME,
+                                               METRICS,
+                                               INSTANCE_TRACKING };
+
+bool
+client_config_sections::can_occur_multiple_times(std::string name)
+{
+    for (client_config_sections section : values) {
+        if (name == section.get_name()) {
+            return section.multiple_occurrence;
+        }
+    }
+    return true;
+}
+
+abstract_dom_config_processor::abstract_dom_config_processor(bool dom_level_3)
+{
+    this->dom_level_3 = dom_level_3;
+    this->strict = true;
+}
+abstract_dom_config_processor::abstract_dom_config_processor(bool dom_level_3,
+                                                             bool strict)
+{
+    this->dom_level_3 = dom_level_3;
+    this->strict = strict;
+}
+
+bool
+abstract_dom_config_processor::matches(std::string config1, std::string config2)
```

Comment:
pass by ref to eliminate copies. similar in most other methods. You can pass by value if you move a variable.

Reply:
should be better now.